esb: Implemented monitor mode (#23323)
nrfconnect/sdk-nrf: samples/esb/esb_monitor/README.rst

.. _esb_monitor:

Enhanced ShockBurst: Monitor
#############################

.. contents::
   :local:
   :depth: 2

> The sample shows how to use the :ref:`ug_esb` protocol in monitor mode.
> It shows how to configure the Enhanced ShockBurst protocol to receive
> all the traffic generated on configured channel and pipes.

Reviewer: ```suggestion
This sample shows how to use the :ref:`ug_esb` protocol in monitor mode.
It shows how to configure the Enhanced ShockBurst protocol to receive all the traffic generated on the configured channel and pipes.
```

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

Overview
********

> The sample consist of one Receiver configured as monitor that uses :ref:`esb_README` library.

Reviewer: ```suggestion
The sample consist of one receiver configured as a monitor that uses the :ref:`esb_README` library.
```

> After building and programming each sample on an nRF54 Series development kit, you can monitor traffic generated by the kits programmed with :ref:`Transmitter <esb_ptx>` and :ref:`Receiver <esb_prx>` respectively.
> Received traffic can be seen in real time using `Serial Terminal app`_.
> Successful monitoring is also indicated by LED changes, which should be in sync on all kits.

Reviewer: ```suggestion
After building and programming each sample on an nRF54 Series DK, you can monitor traffic generated by the kits programmed with the :ref:`Transmitter <esb_ptx>` and :ref:`Receiver <esb_prx>` respectively.
You can see the received traffic in real time using the `Serial Terminal app`_.
Successful monitoring is also indicated by the LEDs that should be in sync on all kits.
```

User interface
***************

> All LEDs:

Reviewer 2: This is only valid if the Monitor is used together with PTX sample.

> All LEDs:
>    Indicate that packets are received.
>    The first four packets turn on the LEDs sequentially.
>    The next four packets turn them off again in the same order.

Reviewer: ```suggestion
All LEDs:

   Indicate that packets are received.
   The first four packets switch on the LEDs sequentially.
   The next four packets switch the LEDs off again in the same order.
```

>    The next four packets turn them off again in the same order.

Reviewer 3: May be worth mentioning that this logic is aligned with the other samples in NCS: esb_prx and esb_ptx as it checks the payloads transmitted by PRX and PTX, which are sequential numbers. It won't work that way in a general case and in theory monitor may listen to more than 2 ESB nodes, then the LED behavior will differ.

Configuration
*************

|config|

Building and running
********************

> The Monitor sample can be found under :file:`samples/esb/esb_monitor` in the |NCS| folder structure.

Reviewer: ```suggestion
The Monitor sample is located in the :file:`samples/esb/esb_monitor` folder in the |NCS| folder structure.
```

> See :ref:`building` and :ref:`programming` for information about how to build and program the application, respectively.

Reviewer: ```suggestion
See :ref:`building` and :ref:`programming` for information on how to build and program the application, respectively.
```

.. include:: /includes/nRF54H20_erase_UICR.txt

FEM support
===========

.. include:: /includes/sample_fem_support.txt

Testing
=======

> After programming kits with Monitor, Transmitter and Receiver samples you can test their functionality.

Reviewer: ```suggestion
After programming the DKs with the Monitor, Transmitter and Receiver samples, you can test their functionality.
```

> Complete the following steps to test the Monitor, Transmitter and Receiver samples:

Reviewer: ```suggestion
Complete the following steps to test the samples:
```

1. Power on all kits.
#. Observe that the LEDs change synchronously on all kits.

> #. Connect to the monitor kit with a terminal emulator (for example, the `Serial Terminal app`_).

Reviewer: ```suggestion
#. Connect to the monitor DK with a terminal emulator (for example, the `Serial Terminal app`_).
```

   See :ref:`test_and_optimize` for the required settings and steps.

> #. Observe the logged traffic for kit programmed with Monitor sample.

Reviewer: ```suggestion
#. Observe the logged traffic of the DK programmed with the Monitor sample.
```

.. _esb_monitor:

Enhanced ShockBurst: Monitor
#############################

.. contents::
   :local:
   :depth: 2

The Monitor sample shows how to use the :ref:`ug_esb` protocol in Monitor mode.
It demonstrates how to configure the Enhanced ShockBurst protocol to receive all the traffic generated on the configured channel and pipes.

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

Overview
********

The sample consist of one receiver configured as a monitor that uses the :ref:`esb_README` library.

> After building and programming each sample on an nRF54 Series DK, you can monitor the traffic generated by the kits programmed with the :ref:`Transmitter <esb_ptx>` and :ref:`Receiver <esb_prx>` samples, respectively.

Author: Not only, other supported boards are listed in requirements section and sample.yaml file
